libcaf_core/caf/policy/work_stealing.hpp

```diff
@@ -85,7 +85,8 @@ class work_stealing : public unprofiled {
       return nullptr;
     }
     // roll the dice to pick a victim other than ourselves
-    size_t victim = d(self).rengine() % (p->num_workers() - 1);
+    std::uniform_int_distribution<size_t> uniform(0, p->num_workers() - 2);
```

Comment:
@mavam I'm not sure if I get what you are trying to get to. We have N workers, but need to choose one of N-1 (any worker other than self).

Now, we get a random number for N-1. If the ID of self is  N, than we have no problem. Otherwise, we exclude the worker holding the last ID but have a change of hit ourselves. The fix is to map our ID to the omitted one... This is straightforward mapping and each ID has the same chance of getting picked while making sure our ID isn't in the pool.

Did I overlook something?